src/auth0-session/handlers/login.ts

```diff
@@ -65,27 +62,31 @@ export default function loginHandlerFactory(
 
     const authParams = {
       ...opts.authorizationParams,
-      nonce: await transientHandler.save('nonce', req, res, transientOpts),
+      nonce: await transientHandler.save('nonce', req, res, { ...transientOpts, value: client.generateRandomNonce() }),
       state: await transientHandler.save('state', req, res, {
         ...transientOpts,
         value: encodeState(stateValue)
       }),
       ...(usePKCE
         ? {
-            code_challenge: transientHandler.calculateCodeChallenge(
-              await transientHandler.save('code_verifier', req, res, transientOpts)
+            code_challenge: await client.calculateCodeChallenge(
+              await transientHandler.save('code_verifier', req, res, {
+                ...transientOpts,
+                value: client.generateRandomCodeVerifier()
+              })
             ),
             code_challenge_method: 'S256'
           }
         : undefined)
     };
 
     const validResponseTypes = ['id_token', 'code id_token', 'code'];
-    assert(
-      validResponseTypes.includes(authParams.response_type as string),
-      `response_type should be one of ${validResponseTypes.join(', ')}`
-    );
-    assert(/\bopenid\b/.test(authParams.scope as string), 'scope should contain "openid"');
+    if (!validResponseTypes.includes(authParams.response_type as string)) {
+      throw new Error(`response_type should be one of ${validResponseTypes.join(', ')}`);
+    }
+    if (!/\bopenid\b/.test(authParams.scope as string)) {
+      throw new Error('scope should contain "openid"');
```

Comment:
Should these errors have a more specific type?

Reply:
This is a configuration error too - we don't have a type for these. We could look at changing that, but it would be a lot of changes for not much/any benefit.